### PR TITLE
Try on Debian

### PR DESCRIPTION
**Pull request by nudgegoonies**
I started creating a Debian Package. There is one bug left, i have to set the architecture in the sources for some reason. I am sure this can be replaced with a variable. Actualle people who want to compile for an architecture other then amd64 have to change line 22 in the makefile.

### CODE REVIEW

**Comment by nudgegoonies** [comment]
My first try was on old Debian Jessie. Now i use Debian Stretch with my own backport of Geany 1.33. This should also run on Buster and Ubuntu (i don't know which versions ships Geany 1.32 or up). I also fixed the problem with the architecture and added a section about how to build the deb package.

File: README.md [hunk 1]
```diff
@@ -36,6 +36,24 @@ This is an [EditorConfig][] plugin for [Geany][].
...
 
 
+### Build Debian/Ubuntu/etc. package from source
+
+1.  Requirements for building the package are a GTK3 Version of Geany >= 1.32.
+    This should work on all Debian like operating systems that meet the
+    requirements in the control file. Further you require `build-essential`
+    and `fakeroot` or `pbuilder` depending on how to build the package.
+
+2.  Run `dpkg-source -b editorconfig-geany` on the sources directory. This
```
Comment (xuhdev) [inline README.md]: Do you mean "the directory where the source tree root resides"?

File: README.md [hunk 2]
```diff
@@ -36,6 +36,24 @@ This is an [EditorConfig][] plugin for [Geany][].
...
+    builds the source package.
+
+3.  Build with dpkg-buildpackage:
+    Run `dpkg-buildpackage -rfakeroot -uc -us` to build inside sources.
```
Comment (xuhdev) [inline README.md]: Can you side "at the root of the source tree"?

Comment (xuhdev) [reply README.md]: Also, do we need step 2 if we use `dpkg-buildpackage`?

File: Makefile [hunk 3]
```diff
@@ -18,10 +18,13 @@
...
 
-EDITORCONFIG_PREFIX = /usr/local
+ifndef EDITORCONFIG_PREFIX
+	EDITORCONFIG_PREFIX = /usr/local
+endif
+TARGET_ARCHITECTURE= $(shell cc -dumpmachine)
 
 CC = cc
-CFLAGS = -Wall -Wextra -O2 -g -fPIC `pkg-config --cflags geany` -I${EDITORCONFIG_PREFIX}/include
+CFLAGS = -Wall -Wextra -O2 -g -fPIC `pkg-config --cflags geany` -I${EDITORCONFIG_PREFIX}/include -I${EDITORCONFIG_PREFIX}/include/geany
```
Comment (xuhdev) [inline Makefile]: Can you explain the reason you change the line here?

Comment (nudgegoonies) [reply Makefile]: That was needed under Pbuilder/Jessie for some Reason. I removed it.

File: README.md (hunk 2 above)
Comment (nudgegoonies) [reply README.md]: You are right. This is only needed for pbuilder that needs the dsc file.

File: README.md (hunk 1 above)
Comment (nudgegoonies) [reply README.md]: Yes, i name it so.

File: README.md (hunk 2 above)
Comment (xuhdev) [reply README.md]: The item numbers are messed here...

File: README.md [hunk 4]
```diff
@@ -36,6 +36,23 @@ This is an [EditorConfig][] plugin for [Geany][].
...
+    This should work on all Debian like operating systems that meet the
+    requirements in the control file. Further you require `build-essential`
+    and `fakeroot` or `pbuilder` depending on how to build the package.
+
+3.  Build with dpkg-buildpackage:
+    Run `dpkg-buildpackage -rfakeroot -uc -us` to build inside sources.
+    Alternatively build with pbuilder:
+    Run `dpkg-source -b DIRECTORYNAME` on the directory where the source tree
+    root resides. This builds the source package, especially the dsc file.
+    Run `sudo pbuilder --build editorconfig-geany_1.dsc` to build. The
```
Comment (xuhdev) [inline README.md]: Is `sudo needed here`?

File: debian/rules [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+# main packaging script based on dh7 syntax
+%:
+	dh $@ 
+
+# debmake generated override targets
+# This is example for Cmake (See http://bugs.debian.org/641051 )
+#override_dh_auto_configure:
+#	dh_auto_configure -- \
+#	-DCMAKE_LIBRARY_PATH=$(DEB_HOST_MULTIARCH)
+
```
Comment (xuhdev) [inline debian/rules]: Can you remove these trailing blank lines?

File: debian/copyright [hunk 6]
```diff
@@ -0,0 +1,29 @@
+Format: http://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
+Upstream-Name: editorconfig-geany
+Source: https://github.com/editorconfig/editorconfig-geany
+
+Files: *
+Copyright: 2012 Hong <hong@topbug.net>
```
Comment (xuhdev) [inline debian/copyright]: This should be `2011-2018 EditorConfig Team`

File: debian/control [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+Standards-Version: 3.9.5
+Homepage: https://github.com/editorconfig/editorconfig-geany
+Vcs-Git: git@github.com:editorconfig/editorconfig-geany.git
+Vcs-Browser: https://github.com/editorconfig/editorconfig-geany
+
+Package: editorconfig-geany
+Architecture: any
+Depends: ${shlibs:Depends}, ${misc:Depends}
+Description: Editorconfig support for Geany
+ Editorconfig support for Geany.
```
Comment (xuhdev) [inline debian/control]: "EditorConfig"

File: README.md (hunk 4 above)
Comment (nudgegoonies) [reply README.md]: Yes, this is requirement of pbuilder. Of course working as root would also work, but this is not recommended in general.

File: debian/rules (hunk 5 above)
Comment (nudgegoonies) [reply debian/rules]: Done.

File: debian/control (hunk 7 above)
Comment (nudgegoonies) [reply debian/control]: You are right.

File: debian/copyright (hunk 6 above)
Comment (nudgegoonies) [reply debian/copyright]: Done.

File: README.md (hunk 2 above)
Comment (nudgegoonies) [reply README.md]: Corrected.

**Comment by nudgegoonies** [comment]
I fixed the remaining issues. No problem. I also want to do it right.

**Pull request merged**